src/storage.ts

```diff
@@ -0,0 +1,266 @@
+
+// API Common
+/**
+ * Utility to generate a reference for the specified file for use with other Storage APIs.
+ * 
+ * @experimental
+ * 
+ * @param key - A file key that will be used to generate the reference.
+ * @returns A StorageObjectReference for the file.
+ */
+declare function getFileReference(key: string): StorageObjectReference;
+
+// API Get
+type DownloadMetadata = {
+  attempts: number;
+  headers: Array<Record<string, string>>;
+  totalRetryDelay: number;
+  statusCode: number;
+  requestId?: string;
+};
+
+type GetURLRequest = {
+  file: StorageObjectReference;
+  expiresIn?: number;
+} & CommonStorageParameters;
+
+type DownloadRequest = {
+  file: StorageObjectReference;
+  onProgress?: (progress: TransferProgress) => void;
+} & CommonStorageParameters;
+
+type DownloadResponse = {
+  content: Blob;
+  contentLength: number;
+  contentType: string;
+  file: StorageObjectReference;
+  metadata: DownloadMetadata;
+};
+
+/**
+ * Generates a pre-signed URL for downloading the specified reference.
+ * 
+ * @remarks
+ * Pre-signed URLs are usable by anyone with the URL, regardless of their access level. This method will not attempt 
+ * to check if the file exists.
+ * 
+ * @throws {@link StorageError} If an error occurs while generating the pre-signed URL.
+ * 
+ * @experimental
+ * 
+ * @param {GetURLRequest} request - A GetURLRequest object.
+ * @returns A promise that will resolve with the pre-signed file URL.
+ */
+declare function getUrl(request: GetURLRequest): Promise<string>;
+
+/**
+ * Downloads the specified object from S3.
+ * 
+ * @throws {@link StorageError} If an error occurs while downloading the file from S3.
+ * 
+ * @experimental
+ * 
+ * @param {DownloadRequest} request - A DownloadFileRequest object.
+ * @returns A promise that will resolve with an object containing the file blob and other information.
+ */
+declare function download(request: DownloadRequest): Promise<DownloadResponse>;
+
+// API List
+type ListFilesRequest = {
+  key: string;
+  pageSize?: PageSize | number;
+  nextToken?: string;
+} & Pick<CommonStorageParameters, 'track' | 'accessLevel' | 'identityId'>;
+
+type ListFilesResponse = {
+  results: Array<StorageObjectReference>;
+  hasNextToken: boolean;
+  nextToken: string;
+};
+
+/**
+ * Lists objects for the specified prefix.
+ * 
+ * @remarks
+ * To list all objects at the specified access level, pass an empty string as the prefix with the request.
+ * 
+ * @throws {@link StorageError} If an error occurs while retrieving the file list.
+ * 
+ * @experimental
+ * 
+ * @param {ListFilesRequest} request - A ListFilesRequest object.
+ * @returns A promise that will resolve with an object containing a list of matching objects from S3.
+ */
+declare function list(request: ListFilesRequest): Promise<ListFilesResponse>;
+
+// API Put
+type PutRequest = {
+  content: string | File | Blob;
+  file: StorageObjectReference;
```

Comment:
I am not sure about this? 

the name is not very descriptive, what if we use `source` and `destination` parameters? similar to `copy`